parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -266,24 +266,24 @@ write_files:
     [plugins."io.containerd.grpc.v1.cri"]
       sandbox_image = "{{GetPodInfraContainerSpec}}"
       [plugins."io.containerd.grpc.v1.cri".containerd]
-        {{ if TeleportEnabled }}
+        {{- if TeleportEnabled }}
         snapshotter = "teleportd"
         disable_snapshot_annotations = false
-        {{ end}}
+        {{- end}}
+        {{ if IsNSeriesSKU }}
         [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime]
           runtime_type = "io.containerd.runtime.v1.linux"
-          {{- if IsNSeriesSKU}}
           runtime_engine = "/usr/bin/nvidia-container-runtime"
-          {{- else}}
-          runtime_engine = "/usr/bin/runc"
-          {{- end}}
         [plugins."io.containerd.grpc.v1.cri".containerd.default_runtime]
           runtime_type = "io.containerd.runtime.v1.linux"
-          {{- if IsNSeriesSKU}}
           runtime_engine = "/usr/bin/nvidia-container-runtime"
-          {{- else}}
-          runtime_engine = "/usr/bin/runc"
-          {{- end}}
+        {{ else}}
+        default_runtime_name = "runc"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          runtime_type = "io.containerd.runc.v2"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted]
+          runtime_type = "io.containerd.runc.v2"
```

Comment:
no runtime_engine  needed for v2? checkin aks-engine settigns.

Reply:
https://github.com/Azure/aks-engine/blob/b3aa75c463764f8fb72b13a7beb4752a6a4e1f37/pkg/api/common/const.go#L283

Reply:
looks like this is the same as aks-engine so I think I'm signed off except for the teleport touch.

Reply:
Yeah I think its best to be explicit here.

Reply:
Has runtime_engine been deprecated?
I don't see it in the readme: https://github.com/containerd/containerd/blob/master/docs/cri/config.md

Reply:
I messed with this today, it only applies to `io.containerd.runtime.v1.linux`... this config is fun.